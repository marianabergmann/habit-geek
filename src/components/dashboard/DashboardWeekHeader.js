import React from "react";

const DashboardWeekHeader = ({
  isCurrentWeek,
  hasPreviousWeek,
  toggleWeek
}) => {
  if (!hasPreviousWeek) return "Just this week";

  return (
    <button
      onClick={() => {
        toggleWeek(!isCurrentWeek);
      }}
    >
      {isCurrentWeek ? "back" : "forward"}
    </button>
  );
};

export default DashboardWeekHeader;
