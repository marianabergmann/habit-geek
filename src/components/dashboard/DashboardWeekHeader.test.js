import React from "react";
import { shallow } from "enzyme";
import DashboardWeekHeader from "./DashboardWeekHeader";

const props = {
  isCurrentWeek: true,
  hasPreviousWeek: true,
  toggleWeek: () => {}
};

describe("DashboardWeekHeader component", () => {
  it("Renders static elements", () => {
    const dashboardWeekHeader = shallow(<DashboardWeekHeader {...props} />);
    expect(dashboardWeekHeader).toMatchSnapshot();
  });

  it("doesn't show buttons to change week when there are no previous weeks", () => {
    const dashboardWeekHeader = shallow(
      <DashboardWeekHeader {...props} hasPreviousWeek={false} />
    );
    expect(dashboardWeekHeader).toMatchSnapshot();
  });

  it("toggles the selected week", () => {
    const toggleWeekMock = jest.fn();
    const dashboardWeekHeader = shallow(
      <DashboardWeekHeader {...props} toggleWeek={toggleWeekMock} />
    );
    dashboardWeekHeader.find("button").simulate("click");
    expect(toggleWeekMock).toHaveBeenCalledWith(false);
  });
});
