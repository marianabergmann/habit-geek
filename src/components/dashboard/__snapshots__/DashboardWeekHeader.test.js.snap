// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardWeekHeader component Renders static elements 1`] = `
<button
  onClick={[Function]}
>
  back
</button>
`;

exports[`DashboardWeekHeader component doesn't show buttons to change week when there are no previous weeks 1`] = `"Just this week"`;
